### ISSUE TITLE
BarChart animation canceled after data change

### ISSUE THREAD

**Issue by RyanGundu** [Abhinandan-Kushwaha/react-native-gifted-charts#517]
Hello!

I'm using the BarChart and providing it data fetched from an API. The Component renders 2 times. Once with the initial state. Then a second time with the updated data from the API. The Animation seems to be canceled but the tooltip renders as if it completed.

```
/ * initial state passed in */
[
    { value: 0, label: 'S' },
    { value: 0, label: 'M' },
    { value: 0, label: 'T' },
    { value: 0, label: 'W' },
    { value: 0, label: 'T' },
    { value: 0, label: 'F', },
    { value: 0, label: 'S' },
]
```

Component snippet: (rendered 2 times)
```
import { BarChart } from 'react-native-gifted-charts';


export const WeekChart = ({ data }) => {
   ...
    return (
      <BarChart
        ...
        animationDuration={500}
        isAnimated
        data={data}
        renderTooltip={...}
      />
    );
};
```

Heres whats happening:

https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/25333009/16f08c26-4466-4ebf-99c8-889903cf1cba

When i remove the animation it renders correctly as shown below:
```
/ * Removed these 2 props */
animationDuration={500}
isAnimated
```

<img width="410" alt="Screenshot 2024-01-24 at 10 30 33 AM" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/25333009/3e75a16e-7078-4567-bb56-526668f7b3ee">

**Comment by RyanGundu** [comment]
The above was on android. This is what it looks like in IOS:

https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/25333009/151078cf-8b62-4573-b50e-e0f78b1eeee2

**Comment by Abhinandan-Kushwaha** [comment]
Hi @RyanGundu 
Is it working as expected in iOS?

**Comment by Abhinandan-Kushwaha** [comment]
Hi @RyanGundu 
I can reproduce the issue on Android. I will release a new version with the fix soon.

For now, I can suggest a work-around. Pass the prop `minHeight={1}`. You can replace 1 with any value you are comfortable with.

**Comment by RyanGundu** [comment]
@Abhinandan-Kushwaha  On IOS its the same issue but it looks like the animation starts then goes back down. I already have `minHeight={2}` set i just removed it from the snippet to reduce clutter. 

I've added this and it seems to be working. Not sure if its actually fixed it though but issue seems to have stopped.

I created this hook:
```
import { useEffect, useRef } from 'react';

export const useIsMounting = () => {
  const isMountingRef = useRef(true);
  useEffect(() => {
    isMountingRef.current = false;
  }, []);
  return isMountingRef.current;
};
```

and used it like this:
```
export const WeekChart = ({ data }) => {
   ...
   const isMounting = useIsMounting();
   
    return (
      <BarChart
        ...
        animationDuration={500}
        isAnimated={!isMounting}
        data={data}
        renderTooltip={...}
        minHeight={2} // was already here
      />
    );
};
```

**Comment by Abhinandan-Kushwaha** [comment]
Hi @RyanGundu 
I have released version `1.4.3`. Please use the latest version of the library.
